bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,154 @@
+            filter_not_found_embed = Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="No words could be found that fit all the filters specified.",
+                color=Colours.soft_red,
+            )
+            await ctx.send(embed=filter_not_found_embed)
+            return
+
+        word = choice(filtered_words)
+
+        user_guess = "_" * len(word)
+        tries = 6
+
+        hangman_embed = Embed(
+            title="Hangman",
+            color=Colours.python_blue,
+        )
+        hangman_embed.set_image(url=IMAGES[tries])
+        hangman_embed.add_field(
+            name=f"The word is `{user_guess}`",
+            value="Guess the word by sending a message with the letter!",
+            inline=False,
+        )
+        original_message = await ctx.send(embed=hangman_embed)
+
+        while user_guess != word:
+            try:
+                message = await self.bot.wait_for(
+                    event="message",
+                    timeout=30.0,
+                    check=lambda msg: msg.author != self.bot if singleplayer == 'm' else msg.author == ctx.author
+                )
+            except TimeoutError:
+                timeout_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description="Looks like the bot timed out! You must send a letter within 60 seconds.",
+                    color=Colours.soft_red,
+                )
+                await ctx.send(embed=timeout_embed)
+                return
+
+            message.content = message.content.lower()
+            if len(message.content) > 1:
+                letter_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description="You can only send a letter for the hangman game!",
+                    color=Colours.soft_red,
+                )
+                await ctx.send(embed=letter_embed)
+                continue
+
+            elif message.content in word:
+                positions = {idx for idx, letter in enumerate(word) if letter == message.content}
+                user_guess = "".join(
+                    [message.content if index in positions else dash for index, dash in enumerate(user_guess)]
+                )
+
+            else:
+                tries -= 1
+
+                if tries <= 0:
+                    losing_embed = Embed(
+                        title="You lost.",
+                        description=f"The word was `{word}`.",
+                        color=Colours.soft_red,
+                    )
+                    await original_message.edit(embed=losing_embed)
+                    return
+
+            hangman_embed.clear_fields()
+            hangman_embed.set_image(url=IMAGES[tries])
+            hangman_embed.add_field(
+                name=f"The word is `{user_guess}`",
+                value="Guess the word by sending a message with the letter!",
+                inline=False,
```

Comment:
Can this also mention through text how many lives there are left?

Reply:
Yeah, that probably makes sense. But don't we have the picture of the man for that?

Reply:
It still may not be clear how many tries you have left.

Reply:
True